### DISCUSSION TITLE
Oauth with SMTP?

### DISCUSSION THREAD

**Discussion by parkinsonr** [nextgenhealthcare/connect#5363] [Q&A]
Is it possible to do Oauth with SMTP? With basic auth no longer being support by microsoft I'm looking for other ways to authenticate with O365

**Comment by jonbartels** [comment]
I think you need an app password - https://support.microsoft.com/en-us/account-billing/using-app-passwords-with-apps-that-don-t-support-two-step-verification-5896ed9b-4263-e681-128a-a6f2979a7944 and https://evermap.com/Tutorial_AMM_AppPasswordOffice365.asp

**Reply by parkinsonr** [reply]
Thanks for the reply, i didnt realize anyone had responded. Looking through this option it sounds like it would still connect using basic auth and username/password?
Client I am working with is requesting Oauth as they do not want a plaintext password being used. Any thoughts on doing this with Mirth?

**Reply by jonbartels** [reply]
Does this help https://github.com/nextgenhealthcare/connect/issues/5508#issuecomment-1321752350 ?

**Reply by parkinsonr** [reply]
Hmm I don't think. I realized i didnt specificy in my OP but I'm trying to send to a microsoft SMTP, not read messages. The options on the SMTP sender destination only have basic auth.

I don't know much about the MSGRAPH API but if there's a POST option to send a message then that could possibly work, although I still don't know how you would authenticate. I see a bearer token is required but not sure how you would go about getting one.
